### PR TITLE
setools: 4.2.0 -> 4.2.2 and fixing build

### PR DESCRIPTION
**Pull request by d-goldin**
###### Motivation for this change

The build was broken because meanwhile setools requires cython
and no bison, swig and flex anymore.

Also, bumping version to newest release, which is not directly related
to the build breakage.

This package is also affected by networkx throwing an exception in some code paths: https://github.com/NixOS/nixpkgs/pull/68727

Related to: https://github.com/NixOS/nixpkgs/issues/68361

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

No known maintainer

### CODE REVIEW

File: pkgs/os-specific/linux/setools/default.nix [hunk 1]
```diff
@@ -8,16 +8,16 @@ with python3.pkgs;
...
     repo = pname;
     rev = version;
-    sha256 = "1bjwcvr6rjx79cdcvaxn68bdrnl4f2a8gnnqsngdxhkhwpddksjy";
+    sha256 = "18kklv26dwm2fdjjzfflvxsq83b2svnwf4g18xq7wsfsri121a90";
   };
 
-  nativeBuildInputs = [ bison flex swig ];
+  nativeBuildInputs = [ cython ];
   buildInputs = [ libsepol ];
   propagatedBuildInputs = [ enum34 libselinux networkx ]
```
Comment (jonringer) [inline pkgs/os-specific/linux/setools/default.nix]:
```suggestion
  propagatedBuildInputs = [ enum34 libselinux networkx setuptools ]
```
```
[nix-shell:~/.cache/nix-review/pr-68728]$ ./results/setools/bin/seinfo --help
Traceback (most recent call last):
  File "/nix/store/l2y3j9bqzd632p1asafxp6cs48ipd2wb-setools-4.2.2/bin/.seinfo-wrapped", line 22, in <module>
    import setools
  File "/nix/store/l2y3j9bqzd632p1asafxp6cs48ipd2wb-setools-4.2.2/lib/python3.7/site-packages/setools/__init__.py", line 76, in <module>
    from .infoflow import InfoFlowAnalysis
  File "/nix/store/l2y3j9bqzd632p1asafxp6cs48ipd2wb-setools-4.2.2/lib/python3.7/site-packages/setools/infoflow.py", line 23, in <module>
    import networkx as nx
  File "/nix/store/m0nlk8hwb1fd1dn1aaiwv09wqjmcgyxq-python3.7-networkx-2.2/lib/python3.7/site-packages/networkx/__init__.py", line 128, in <module>
    import networkx.drawing
  File "/nix/store/m0nlk8hwb1fd1dn1aaiwv09wqjmcgyxq-python3.7-networkx-2.2/lib/python3.7/site-packages/networkx/drawing/__init__.py", line 6, in <module>
    from . import nx_pydot
  File "/nix/store/m0nlk8hwb1fd1dn1aaiwv09wqjmcgyxq-python3.7-networkx-2.2/lib/python3.7/site-packages/networkx/drawing/nx_pydot.py", line 27, in <module>
    from pkg_resources import parse_version
ModuleNotFoundError: No module named 'pkg_resources'
```

Comment (d-goldin) [reply pkgs/os-specific/linux/setools/default.nix]: Its linked in the related issue, this is within networkx.

Comment (d-goldin) [reply pkgs/os-specific/linux/setools/default.nix]: This one: https://github.com/NixOS/nixpkgs/pull/68727
In reality this probably goes deeper, as the networkx packaged does not actually provide either pygraphviz or pydot, so likely the networkx package needs to be improved a little bit.

**Review by worldofpeace — APPROVED** [review]
Tested with networkx PR and builds and executes without an import error.

**Comment by worldofpeace** [comment]
backported in 28e5cee0478871ad8e5aa1c6a958c8d8eb177a99

**Pull request merged**